### ISSUE TITLE
Build failure on OpenIndiana

### ISSUE THREAD

**Issue by rubenk** [grke/burp#351]
```
$ make
Making all in src
make[1]: Entering directory '/home/ruben/src/burp/src'
make  all-am
make[2]: Entering directory '/home/ruben/src/burp/src'
  CC       burp-alloc.o
  CC       burp-asfd.o
  CC       burp-async.o
  CC       burp-attribs.o
  CC       burp-base64.o
  CC       burp-berrno.o
  CC       burp-bfile.o
  CC       burp-bu.o
  CC       burp-cmd.o
  CC       burp-cntr.o
  CC       burp-conf.o
  CC       burp-conffile.o
In file included from conffile.c:6:0:
lock.h:11:21: error: conflicting declaration ‘typedef struct lock lock_t’
 typedef struct lock lock_t;
                     ^
In file included from /usr/include/sys/machtypes.h:36:0,
                 from /usr/include/sys/types.h:41,
                 from /usr/include/sys/stat.h:35,
                 from /usr/include/fcntl.h:37,
                 from burp.h:22,
                 from conffile.c:1:
/usr/include/ia32/sys/machtypes.h:65:23: note: previous declaration as ‘typedef unsigned char lock_t’
 typedef unsigned char lock_t;  /* lock work for busy wait */
                       ^
Makefile:1328: recipe for target 'burp-conffile.o' failed
make[2]: *** [burp-conffile.o] Error 1
make[2]: Leaving directory '/home/ruben/src/burp/src'
Makefile:583: recipe for target 'all' failed
make[1]: *** [all] Error 2
make[1]: Leaving directory '/home/ruben/src/burp/src'
Makefile:431: recipe for target 'all-recursive' failed
make: *** [all-recursive] Error 1
```

**Comment by rubenk** [comment]
We should probably just get rid of the typedef?

**Comment by rubenk** [comment]
Fixed by 92ccc1f
